### PR TITLE
Dev/packs

### LINKED ISSUE #35: Pack object
- [x] getSetList
- [x] probabilites
- [x] setList

### LINKED ISSUE #56: Add Users as a Property of Client
Once this is done, modify the backend service methods to get all users in the maui frontend.

### PR DESCRIPTION
**Pull request by idafum**
Closes #56 
Closes #35

### CODE REVIEW

File: CombatCrittersSharp/Client.cs [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
-    public IRest Rest { get; } = new Rest(apiUri);
+    public IRest Rest { get; }
     public IUser? User { get; private set; }
+    public UserManager? Users { get; private set; }
 
+    //constructor
+    public Client(string apiUri)
+    {
+        Rest = new Rest(apiUri);
+    }
```
Comment (VanJackal) [inline CombatCrittersSharp/Client.cs]: what was wrong with the existing instantiation?

File: CombatCrittersSharp/objects/pack/IPack.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace CombatCrittersSharp.objects.pack
+{
+    public interface IPack
+    {
+        string Image { get; }
+        string Name { get; }
+        int PackId { get; }
+
+        List<ICard>? Contents { get; } // nullable to indicate it may not be initially loaded
```
Comment (VanJackal) [inline CombatCrittersSharp/objects/pack/IPack.cs]: should be an async getter instead

File: CombatCrittersSharp/rest/payloads/PackPayload.cs [hunk 3]
```diff
@@ -1,29 +1,8 @@
...
-
-public record PackCardSlotItem(
-    int rarity, // rarity id
-    int weight // weights for each rarity will be summed, 
-               //the proportion of the total will be the probablity of getting a card with that rarity missing rarities will assume a weight of 0
-);
-
-public record PackContentsPayload(
-    CardPayload[] cards
-);
```
Comment (VanJackal) [inline CombatCrittersSharp/rest/payloads/PackPayload.cs]: why are these all being removed?

File: CombatCrittersSharp/rest/payloads/PackPayload.cs [hunk 4]
```diff
@@ -1,29 +1,8 @@
...
 namespace CombatCrittersSharp.rest.payloads;
 
-public record PackPayload(
-    string name,
-    string image,
-    int packid
+public record PackDetailsPayload(
+    string Name, //Name of pack
+    string Image, //Pack Image
+    int Packid //Pack Id
```
Comment (VanJackal) [inline CombatCrittersSharp/rest/payloads/PackPayload.cs]: These cannot be renamed, they are conforming to the api, the record name `PackPayload` is also reflecting api but is less vital

File: CombatCrittersSharp/managers/interfaces/IPackManager.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        /// </summary>
+        /// <returns></returns>
+        Task<List<Pack>> GetAllPacksAsync();
+
+        /// <summary>
+        /// Retrieves the contents (cards) of a specific pack by its ID.
+        /// </summary>
+        /// <param name="packId"></param>
+        /// <returns></returns>
+        Task<List<ICard>> GetPackContentsAsync(int packId);
```
Comment (VanJackal) [inline CombatCrittersSharp/managers/interfaces/IPackManager.cs]: should be on IPack

File: CombatCrittersSharp/managers/PacksManager.cs
Comment (VanJackal) [inline CombatCrittersSharp/managers/PacksManager.cs:1]: AuthExceptions should only be used if there is an Authentication issue (HTTP 403). The rest exceptions should either just not be handled here or should be wrapped with a more appropriate exception

**Review by whatdoes3plus1equalsto — APPROVED** [review]

**Review by VanJackal — CHANGES_REQUESTED** [review]

File: CombatCrittersSharp/rest/payloads/PackPayload.cs (hunk 3 above)
Comment (VanJackal) [reply CombatCrittersSharp/rest/payloads/PackPayload.cs]: (they should probably be in seperate files tbh but they are needed)

File: CombatCrittersSharp/objects/pack/Pack.cs
Comment (VanJackal) [inline CombatCrittersSharp/objects/pack/Pack.cs:1]: probably dont need a local copy of the pack contents

**Review by VanJackal — APPROVED** [review]

File: CombatCrittersSharp/objects/pack/Pack.cs
Comment (idafum) [reply CombatCrittersSharp/objects/pack/Pack.cs:1]: okay

**Pull request merged**